rust/src/client.rs

```diff
@@ -100,103 +104,91 @@ impl Client {
         }
     }
 
-    /// [`Client`] duties within a round
     pub async fn during_round(&mut self) -> Result<Task, ClientError> {
+        debug!(client_id = %self.id, "polling for new round parameters");
         loop {
-            if let Some(round_params_data) = self.handle.get_round_parameters().await {
-                if let Some(ref round_params) = round_params_data.round_parameters {
-                    if round_params.pk != self.coordinator_pk {
-                        // new round: save coordinator pk
-                        self.coordinator_pk = round_params.pk;
-                        debug!(client_id = %self.id, "computing sigs and checking task");
-                        let round_seed = round_params.seed.as_slice();
+            if let Some(params_outer) = self.proxy.get_params().await? {
+                if let Some(params_inner) = params_outer.round_parameters {
+                    // new round?
+                    if params_inner.pk != self.coordinator_pk {
```

Comment:
Here you can also check whether round_seed has changed.

Reply:
we could, but I'd prefer to not track the round seed if we don't (strictly speaking) need to - comparing the coordinator public key should in theory be sufficient.